numpy/linalg/tests/test_linalg.py

```diff
@@ -289,29 +319,35 @@ def _stride_comb_iter(x):
             xi = np.lib.stride_tricks.as_strided(x, strides=s)
             yield xi, "stride_xxx_0_0"
 
-for src in (SQUARE_CASES,
-            NONSQUARE_CASES,
-            HERMITIAN_CASES,
-            GENERALIZED_SQUARE_CASES,
-            GENERALIZED_NONSQUARE_CASES,
-            GENERALIZED_HERMITIAN_CASES):
-
+def _make_strided_cases():
     new_cases = []
-    for case in src:
-        for a, a_tag in _stride_comb_iter(case.a):
-            for b, b_tag in _stride_comb_iter(case.b):
-                new_case = LinalgCase(case.name + "_" + a_tag + "_" + b_tag, a, b,
-                                      exception_cls=case.exception_cls)
+    for case in CASES:
+        for a, a_label in _stride_comb_iter(case.a):
+            for b, b_label in _stride_comb_iter(case.b):
+                new_case = LinalgCase(case.name + "_" + a_label + "_" + b_label, a, b,
+                                      tags=case.tags | {'strided'})
                 new_cases.append(new_case)
-    src.extend(new_cases)
+    return new_cases
+
+CASES += _make_strided_cases()
 
 
 #
 # Test different routines against the above cases
 #
 
-def _check_cases(func, cases):
-    for case in cases:
+def _check_cases(func, require=set(), exclude=set()):
+    """
+    Run func on each of the cases with all of the tags in require, and none
+    of the tags in exclude
+    """
+    for case in CASES:
+        # filter by require and exclude
+        if case.tags & require != require:
```

Comment:
Hmm, this is a bit clumsy for a lookup, although I suspect it would require a redesign to improve it. Probably not worth the effort.